iceoryx_hoofs/source/posix_wrapper/mutex.cpp

```diff
@@ -16,17 +16,171 @@
+            switch (result.get_error().errnum)
+            {
+            case EPERM:
+                LogError() << "Unsufficient permissions to set the mutex priority ceiling.";
+                return cxx::error<MutexError>(MutexError::PERMISSION_DENIED);
+            case ENOSYS:
+                LogError() << "The platform does not support mutex priority ceiling.";
+                return cxx::error<MutexError>(MutexError::PRIORITIES_UNSUPPORTED_BY_PLATFORM);
+            case EINVAL:
+            {
+                auto minimumPriority = getSchedulerPriorityMinimum(Scheduler::FIFO);
+                auto maximumPriority = getSchedulerPriorityMaximum(Scheduler::FIFO);
+
+                LogError() << "The priority ceiling \"" << m_priorityCeiling
+                           << "\" is not in the valid priority range [ " << minimumPriority << ", " << maximumPriority
+                           << "] of the Scheduler::FIFO.";
+                return cxx::error<MutexError>(MutexError::INVALID_PRIORITY_CEILING_VALUE);
+            }
+            }
+        }
+    }
+
+
+    result = posixCall(pthread_mutexattr_setrobust)(&mutexAttributes, static_cast<int>(m_threadTerminationBehavior))
+                 .returnValueMatchesErrno()
+                 .evaluate();
+    if (result.has_error())
+    {
+        LogError() << "This should never happen. An unknown error occurred while setting up the mutex thread "
+                      "termination behavior.";
+        return cxx::error<MutexError>(MutexError::UNDEFINED);
+    }
+
+
+    uninitializedMutex.emplace();
+    result = posixCall(pthread_mutex_init)(&uninitializedMutex->m_handle, &mutexAttributes)
+                 .returnValueMatchesErrno()
+                 .evaluate();
+    if (result.has_error())
+    {
+        uninitializedMutex->m_isDescructable = false;
+        uninitializedMutex.reset();
+
+        switch (result.get_error().errnum)
+        {
+        case EAGAIN:
+            LogError() << "Not enough resources to initialize another mutex.";
+            return cxx::error<MutexError>(MutexError::INSUFFICIENT_RESOURCES);
+        case ENOMEM:
+            LogError() << "Not enough memory to initialize mutex.";
+            return cxx::error<MutexError>(MutexError::INSUFFICIENT_MEMORY);
+        case EPERM:
+            LogError() << "Unsufficient permissions to create mutex.";
+            return cxx::error<MutexError>(MutexError::PERMISSION_DENIED);
+        default:
+            LogError() << "This should never happen. An unknown error occurred while initializing the mutex handle. "
+                          "This is possible when the handle is an already initialized mutex handle.";
+            return cxx::error<MutexError>(MutexError::UNDEFINED);
+        }
+    }
+
+    uninitializedMutex->m_isDescructable = true;
+    return cxx::success<>();
+}
+
+/// @todo iox-#1036 remove this, introduced to keep current API temporarily
+Mutex::Mutex(bool f_isRecursive) noexcept
```

Comment:
This constructor will be removed when the mutex builder is integrated into iceoryx.